pkg/operator/operator.go

```diff
@@ -213,3 +219,41 @@ func getVnetGUID(cfg *auth.Config, subnetID string) (string, error) {
 	}
 	return *vnet.Properties.ResourceGUID, nil
 }
+
+// WaitForCRDs waits for the required CRDs to be available with a timeout
+func WaitForCRDs(ctx context.Context, client *discovery.DiscoveryClient, timeout time.Duration, log logr.Logger) error {
+	var requiredCRDs = []schema.GroupVersionResource{
```

Comment:
I think you could avoid hardcoding these and instead just reference the scheme?

https://github.com/kubernetes-sigs/karpenter/blob/93c0ef3a4d02bfc1124533a5661d19b9373dfb89/pkg/apis/v1/doc.go#L32

and 

for us at: pkg/apis/v1alpha2/doc.go

There's likely other stuff in scheme too, but you could filter to the `karpenter.sh` and `karpenter.azure.com` groups via `scheme.Scheme.ObjectKinds()` (though note that returns `Kind` rather than `Resource` so the format changes a bit, but it accomplishes the same thing I think.

Reply:
Alternatively you could possibly leave it hardcoded here but write a test (possibly using the scheme above) that ensures we don't have new versions and/or new resources sneak in but not wait for them?

Reply:
Take a look at the version with RESTMapper

Reply:
Updated RESTMapper is nice but doesn't actually solve what I was getting at here which was: we're hardcoding the CRD list here that we want to watch which means we need to update it when a new CRD (or a new version) is added, and that would be easy to miss because it would _mostly_ work (since CRDs are generally applied together) but would be incorrect.

Sourcing from scheme with a filter, or having a test, protects against future evolution where we add CRDs or versions and forget to add them here.

Reply:
Agree on the value, ran into some minor issues when filtering from Scheme, for expediency will merge this for now and circle back to this later.